models/repo_watch.go

```diff
@@ -4,42 +4,118 @@
 
 package models
 
-import "fmt"
+import (
+	"fmt"
+
+	"code.gitea.io/gitea/modules/setting"
+)
+
+// RepoWatchMode specifies what kind of watch the user has on a repository
+type RepoWatchMode int8
+
+const (
```

Comment:
can we add a more general watch enum ?

so models.issue_watch can ue them too

Reply:
I guess I can update this PR if you want to base from it, no problem. I've only pointed this out for you in case you wanted to use the code as an example for yours (lots of things are resolved, and they needed some thinking). But if you base from this, you'll need to wait until it's merged to resume yours, I think. Give me a couple of days and I'll get this ready.

Reply:
@6543 updated.

I didn't make the enum more general because it will take some work to generalize the corresponding functions as well. Generalizing the enum is really the easy part. 😉